### DISCUSSION TITLE
Missing Data for Latter Parts of Experiment

### DISCUSSION THREAD

**Discussion by ssweet4** [jspsych/jsPsych#2524] [Q&A]
Hello!

We are currently trying to run an experiment using Pavlovia and jsPsych. We have run a number of iterations of this experiment, which all have slight adjustments but are otherwise relatively similar. This current one we are trying to run is also similar to the last one we ran, which had no issues with recording all of the data. The current study's github repository is here:

https://github.com/ssweet4/SL-of-SWS_exp-1d_nonspeech_Stekelenburg_Induction

The issue we are having is that when the experiment is run, only the data from the first 2/3 of the experiment is recorded (i.e. out of the four tasks in the experiment, the data stops being recorded after the 19-24th trial out of the 30 trials that make up the third task). We are uncertain as to what is causing this, as the data stops being recorded around a similar point in the task, but not at one certain trial, sometimes stopping at trial 20, other times at 22, and so on. Additionally, this third task was not altered from the previous iteration of the experiment.

Moreover, on some days when I run through it, all of the data is recorded. However, the next day it may be back to only recording 2/3 of the data.

One final thing of note is that although Pavlovia shows the date of the last session, it also shows both the completed and aborted number of sessions as 0. I am unsure if this is relevant to the current issue, however.

I have read in other posts that data loss when using Pavlovia / jsPsych can occur due to participants exiting out of the experiment quickly; however, I have run the study and waited on the final landing page of the study for 5-10 minutes, and this issue still persists. I wondered if perhaps it was due to our code using an older version of jsPsych (6.2.0); however, when I recently ran through a previous iteration of the experiment that also used this older version of jsPsych, it was able to still record all of the data.

To summarize, sometimes the csv files are missing the data from the latter part of the experiment. Any possible insight or suggestions on what may be causing this would be very much appreciated. Thank you in advance!

**Comment by jodeleeuw** [comment]
Hi @ssweet4,

It looks like you are using jQuery to post the data to a server hosted on pythonanywhere.com.  My recommendation would be to troubleshoot the POST request and ensure that the data leaving match the data that are being received on your server. If they do and the CSV string doesn't contain the full set of data then the issue is probably something to do with jsPsych's CSV encoder. But we haven't had any reports of an issue with that before, so I'd want to rule out any issues on the communication or backend side first.

Another option would be to switch to JSON strings instead of CSV to see if that fixes the issue. That would certainly suggest a problem with the CSV encoder.
